api/routes/comments/comments.controller.js

```diff
@@ -0,0 +1,55 @@
+const express = require("express");
+const router = express.Router();
+const { createComment } = require("./comments.service");
+const { authenticateUser } = require("../users/users.service");
+const {
+  validateRequiredPostFields,
+  handleError,
+} = require("../../utils/requestHelper");
+
+const crypto = require("crypto");
+
+router.post("/posts/:postId/comments", async (req, res) => {
+  const { postId } = req.params;
+  const { body } = req;
+
+  const validationErrors = validateRequiredPostFields(
+    {
+      postId: {
+        required: true,
+        notEmpty: true,
+      },
+      text: {
+        required: true,
+        notEmpty: true,
+      },
+      userId: {
+        required: true,
+        notEmpty: true,
+      },
+      secret: {
+        required: true,
+        notEmpty: true,
+      },
+    },
+    body
+  );
+
+  if (validationErrors.error) {
+    res.status(400).json(validationErrors);
+    return;
+  }
+
+  const isValid = await authenticateUser(body.userId, body.secret);
+
+  if (isValid) {
+    const uuid = crypto.randomUUID();
```

Comment:
please create the uuid in the service. i would consider this business logic so it belongs in service. controller validates and handles request/response only